src/user_crate/state_generated.rs

```diff
@@ -70,41 +70,49 @@ impl StateGenerated {
         let (user_code, user_dependencies) = parse_source_and_deps(
             &String::from_datum(prosrc_datum, is_null).ok_or(PlRustError::NullSourceCode)?,
         )?;
-        let argnames_datum = pg_sys::SysCacheGetAttr(
-            pg_sys::SysCacheIdentifier_PROCOID as i32,
-            proc_tuple,
-            pg_sys::Anum_pg_proc_proargnames as pg_sys::AttrNumber,
-            &mut is_null,
-        );
-        let argnames = Vec::<Option<_>>::from_datum(argnames_datum, is_null);
-
-        let argtypes_datum = pg_sys::SysCacheGetAttr(
-            pg_sys::SysCacheIdentifier_PROCOID as i32,
-            proc_tuple,
-            pg_sys::Anum_pg_proc_proargtypes as pg_sys::AttrNumber,
-            &mut is_null,
-        );
-        let argtypes = Vec::<_>::from_datum(argtypes_datum, is_null).unwrap();
 
         let proc_entry = PgBox::from_pg(pg_sys::heap_tuple_get_struct::<pg_sys::FormData_pg_proc>(
             proc_tuple,
         ));
 
-        let mut argument_oids_and_names = Vec::new();
-        for i in 0..proc_entry.pronargs as usize {
-            let type_oid = argtypes.get(i).expect("no type_oid for argument");
-            let name = argnames.as_ref().and_then(|v| v.get(i).cloned()).flatten();
+        let return_oid = PgOid::from(proc_entry.prorettype);
+
+        let variant = match return_oid == pgx::PgBuiltInOids::TRIGGEROID.oid() {
+            true => CrateVariant::trigger(),
+            false => {
+                let argnames_datum = pg_sys::SysCacheGetAttr(
+                    pg_sys::SysCacheIdentifier_PROCOID as i32,
+                    proc_tuple,
+                    pg_sys::Anum_pg_proc_proargnames as pg_sys::AttrNumber,
+                    &mut is_null,
+                );
+                let argnames = Vec::<Option<_>>::from_datum(argnames_datum, is_null);
+
+                let argtypes_datum = pg_sys::SysCacheGetAttr(
+                    pg_sys::SysCacheIdentifier_PROCOID as i32,
+                    proc_tuple,
+                    pg_sys::Anum_pg_proc_proargtypes as pg_sys::AttrNumber,
+                    &mut is_null,
+                );
+                let argtypes = Vec::<_>::from_datum(argtypes_datum, is_null).unwrap();
+
+                let mut argument_oids_and_names = Vec::new();
+                for i in 0..proc_entry.pronargs as usize {
+                    let type_oid = argtypes.get(i).expect("no type_oid for argument");
+                    let name = argnames.as_ref().and_then(|v| v.get(i).cloned()).flatten();
+
+                    argument_oids_and_names.push((PgOid::from(*type_oid), name));
+                }
```

Comment:
It would be nice if we could wrap this kind of thing in a Rust projection before it "got this far" into the PL/Rust code, but this was mostly already here.